markdown.dtx

```diff
@@ -23699,12 +23703,25 @@ function M.reader.new(writer, options)
                      * Cb("level")
                      / writer.heading
 
-  parsers.SetextHeading = #(parsers.line * S("=-"))
-                        * Ct(parsers.linechar^1
-                            / self.parser_functions.parse_inlines)
-                        * parsers.newline
+  function remove_enclosing_whitespaces(o)
+    local heading_content = ""
+    for line in o:gmatch("([^\n]*)\n?") do
+      heading_content = heading_content .. (line:gsub("^%s*(.-)%s*$", "%1" .. "\n"))
+    end
+    return heading_content
+  end
+
+  parsers.heading_line  = parsers.line 
+                        - parsers.blankline 
+                        - parsers.thematic_break_lines
+
+  parsers.heading_text = parsers.heading_line
+                       * (parsers.heading_line - parsers.heading_level)^0
+
+  parsers.SetextHeading = #(parsers.heading_text * parsers.heading_level)
+                        * Ct(parsers.heading_text / remove_enclosing_whitespaces
+                            / self.parser_functions.parse_blocks_nested )
```

Comment:
Would not using `self.parser_functions.parse_blocks_nested` violate https://github.com/lostenderman/markdown/issues/89? Why not use `self.parser_functions.parse_inlines`?